contrib/zstd/utils/utils_zstd.cc

```diff
@@ -67,8 +67,13 @@ absl::Status DecompressInMemory(ZstdApi& api, std::ifstream& in_stream,
     return absl::UnavailableError("Unable to read file");
   }
 
+  // To not transfer memory twice (for getFrameContentSize and decopmress),
+  // tranfer memory before using it.
+  SAPI_RETURN_IF_ERROR(api.GetSandbox()->Allocate(&inbuf, true));
+  SAPI_RETURN_IF_ERROR(api.GetSandbox()->TransferToSandboxee(&inbuf));
+
```

Comment:
This is actually what `inbuf.PtrBefore()` is doing. I think we just need the change on line 88 below.

Reply:
I did that for a purpose. 
I think that:
```
api.ZSTD_getFrameContentSize(
                                         inbuf.PtrBefore(), inbuf.GetSize()));
                                         
[a lot of code]
api.ZSTD_decompress(outbuf.PtrAfter(), size,
                                         inbuf.PtrNone(), inbuf.GetSize()));
```

Won't be obvious for the reader, the reader may be wondering what happen, why the memory isn't synchronize in the second case.
I think something like:
```
inbuf.TransferNow();

api.ZSTD_getFrameContentSize(
                                         inbuf.PtrNone(), inbuf.GetSize()));
                                         
[a lot of code]
api.ZSTD_decompress(outbuf.PtrAfter(), size,
                                         inbuf.PtrNone(), inbuf.GetSize()));
```
Is more readable, and intention is obvious. So I would prefer to have method `TransferNow` which may even wrap the PtrBefore but doesn't return the ptr.
But this is a style issue, so I won't argue intensively on this, so its yours decision. :)

Reply:
And one more think: if somebody will decide that they don't need first call to the function (in this case `ZSTD_getFrameContentSize`), the may not noticed that the rest of code depends on memory synchronization hidden in the first call.

Reply:
Regarding style: Ideally, we want to get rid of any manual calls to `TransferToSandboxee()` and similar. I am working on a better API for this.

For the current review: I'd prefer to change it back to just `inbuf.PtrBefore()`. If there is anything that is not obvious, it should have a short comment (e.g. "PtrBefore() allocates and transfers the buffer")

Reply:
Ok.